File: binarySearch/35_searchInsert.go
package binarySearch

func searchInsert(nums []int, target int) int {
	left, right := 0, len(nums)-1
	ans := len(nums)
	for left <= right {
		mid := left + (right-left)/2
		if nums[mid] <= target {
			ans = mid
			left = mid + 1
		} else if nums[mid] > target {
			right = mid - 1
		}

	}
	return ans
}

File: slice/35_search_insert.rs
impl Solution {
    pub fn search_insert(nums: Vec<i32>, target: i32) -> i32 {
        // 二分查找
        if nums.is_empty() {
            return 0;
        }

        let (mut left, mut right) = (0, nums.len() - 1);

        while left <= right {
            let mid = left + (right - left) / 2;
            match nums[mid].cmp(&target) {
                std::cmp::Ordering::Less => left = mid + 1,
                std::cmp::Ordering::Greater => right = mid - 1,
                std::cmp::Ordering::Equal => return mid as i32,
            }
        }
        return left as i32;
    }
}